include/ceed/types.h

```diff
@@ -23,7 +33,7 @@
 #ifndef CEED_QFUNCTION
 #define CEED_QFUNCTION(name) \
   static const char name ## _loc[] = __FILE__ ":" #name;        \
-  static int name
+  CEED_QFUNCTION_ATTR static int name
```

Comment:
If we want to use intel's `#pragma forceinline`, we'd need to do something like this:
```suggestion
  CEED_QFUNCTION_ATTR 
  static int name
```

I'm not super familiar with C/C++ preprocessing, so I'm not sure if this form is compatible with `__attribute__` and  `_Pragma`.

Reply:
I think this should be fine, but you would need to write
```

  CEED_QFUNCTION_ATTR                                           \
  static int name
```

Reply:
Would that require a change to the python code fix?

Reply:
Right, you should be fine switching
```

Reply:
Yeah, that's what I ended up with.

Reply:
Turns out that putting `CEED_QFUNCTION_ATTR` on a separate line both doesn't do anything and isn't needed.

It doesn't do anything because apparently the preprocessor removes the `\` before it does anything else. They allow for line continutation, but don't affect the actual output.

It's not needed because a `_Pragma()` will automatically insert a newline.

This is confirmed from looking at the output of gcc after preprocessing. Keeping `CEED_QFUNCTION_ATTR` on the same physical line in the macro and setting it equal to `__attribute__((flatten))` gets:
```c
__attribute__((flatten)) static inline State StateFromY(NewtonianIdealGasContext gas,
                                       const CeedScalar Y[5], const CeedScalar x[3]) {
```

while setting it to `_Pragma("message(__FILE__)")` gets:
```c
# 144 "/home/jrwrigh/software/libCEED/examples/fluids/problems/../qfunctions/newtonian_state.h"
#pragma message("/home/jrwrigh/software/libCEED/examples/fluids/problems/../qfunctions/newtonian_state.h")
# 144 "/home/jrwrigh/software/libCEED/examples/fluids/problems/../qfunctions/newtonian_state.h"
 static inline State StateFromY(NewtonianIdealGasContext gas,
                                       const CeedScalar Y[5], const CeedScalar x[3]) {
```